### ISSUE TITLE
Reproducibility on ScanObjectNN

### ISSUE THREAD

**Issue by Siyeong-Lee** [FabianFuchsML/se3-transformer-public#30]
Hi, Fabian!

I am experimenting with ScanObjectNN by setting the output channel of the model to 15 for the QM9 dataset and changing the MAE loss to cross-entropy. Additionally, the batch size is set to 4 and the learning rate is modified accordingly.

For the input values, I set the edge features to {} and the node features to {0: normalized positions of point}, and used the graph generated by KNN (k=10). Under these settings, I am not able to restore the paper's experimental results.

Please let me know if there is anything I should be concerned about.

Thank you.

**Comment by FabianFuchsML** [comment]
Hi Siyeong-Lee,

Happy to hear that you are interested in our work!

The relevant sections in the paper are 4.2 and D.1 in the appendix, where we discuss a number of details. Particularly important is the bit about symmetry breaking. This is an artefact of object alignment with the gravitational axis and the depth dimension of rgbd cameras.

> the batch size is set to 4

This seems quite low to me and might hurt performance. We used batch size 10.

> For the input values, I set the edge features to {}

I would advice to use the distances between pairs of points as edge features.

In general, bear in mind that this repository is a reimplementation of the original code for IP reasons. Hence, optimal hyperparameters might very well be different depending on how accurately we managed to re-implement. E.g. differences in parameter initialisations might mean different optimal learning rates etc.

However, since Nvidia published the accelerated version, I would recommend anyone to use their version of the code. They managed to speed up training of the SE(3)-Transformer by up to 21 times and reduced memory consumption by up to 43 times, which is obviously amazing. This will allow you to experiment much more quickly and try out common tricks in point cloud classification such as data augmentation, pre-training, ensembling, etc…

https://developer.nvidia.com/blog/accelerating-se3-transformers-training-using-an-nvidia-open-source-model-implementation/

Best,
Fabian

**Comment by Siyeong-Lee** [comment]
Thank you for detailed explanation and quick reply.
Based on your comments, the results of the experiment are as follows.

1. Although batch size is set to 10, there is no additional improvements.
2. As you recommended, I experimented using the nvidia version. Since there is a module  that adds the distance between pairs of points to the edge feature in the nvidia version, there is also no additional performance improvement.

Judging from the experiments above, I agree with you that symmetry breaking is important.
For this, I changed the node feature {'0': normalized positions for x, y and rgb for points, '1': normalized positions for z}. Does this method match what you intended?

``` python
# args.node_feature_dim[0] = 5 (RGBxy)
# args.node_feature_dim[1] = 1 (z)
# args.num_degrees = 4
# args.num_channels = 64
# args.edge_feature_dim = 1

model = PointSE3Transformer(
        fiber_in=Fiber({0: args.node_feature_dim[0], 1: args.node_feature_dim[1]}),
        fiber_out=Fiber({0: args.num_degrees * args.num_channels}),
        fiber_edge=Fiber({0: args.edge_feature_dim}),
        **vars(args))
```

**Comment by Siyeong-Lee** [comment]
Thank you for the reply. Your comments have been very helpful.
